packages/lib/src/interactions/interaction.ts

```diff
@@ -0,0 +1,44 @@
+import { castArray } from 'lodash';
+import type { ModifierKey } from 'react';
+
+import type { InteractionConfig } from './models';
+import { MouseButton } from './models';
+
+export class Interaction {
+  public readonly buttons: MouseButton[];
+  public readonly modifierKeys: ModifierKey[];
+  public readonly isWheel: boolean;
```

Comment:
Note that I'm no longer storing `'Wheel'` in `buttons` like it was done in `InteractionsProvider`. It makes the constructor a bit more verbose, but it clarifies the matching logic a little.

The `InteractionConfig` still requires providing `'Wheel'` through the `button` prop for convenience but by design, it still disallows `[MouseButton.Left, 'Wheel']`, for instance.